core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```diff
@@ -277,10 +311,20 @@ public long cleanUpAllAllocatedMemory() {
         freedBytes += memory.size();
         // We don't call free() here because that calls Set.remove, which would lead to a
         // ConcurrentModificationException here.
-        executorMemoryManager.free(memory);
+        memoryManager.tungstenMemoryAllocator().free(memory);
```

Comment:
can these just be `memoryManager.free(memory)`? i.e. We still keep the actual allocation code in a separate class, but just provide a helper method to call the internal memory allocator. Then you don't need to say `tungstenMemoryAllocator` everywhere.